optuna/pruners/_percentile.py

```diff
@@ -33,19 +33,20 @@ def _get_percentile_intermediate_result_over_trials(
     if len(completed_trials) == 0:
         raise ValueError("No trials have been completed.")
 
-    if direction == StudyDirection.MAXIMIZE:
+    intermediate_values = [
+        t.intermediate_values[step] for t in completed_trials if step in t.intermediate_values
+    ]
+    is_maximize = direction == StudyDirection.MAXIMIZE
```

Comment:
This is a small detail but this local variable doesn't really improve the code quality and it's somewhat unrelated to the fix. Might it be sensible to revert this change?

Reply:
Sure! I forgot to revert this after I returned `math.nan` 🤣